contracts/Pausable.sol

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * @notice Emitted when the address authorized to pause/unpause the contract is
+     * changed.
+     * @param pauser THe new pauser.
+     */
+    event PauserSet(address pauser);
+
+    /**
+     * @notice Used when an `onlyWhenNotPaused` function is called while the
+     * contract is paused.
+     */
+    error Paused();
+
+    /**
+     * @notice Used when an `onlyPauser` function is called with a different
+     * address.
+     */
+    error OnlyPauser();
+
+    /**
+     * @notice Reverts if the contract is paused.
+     */
+    modifier onlyWhenNotPaused() {
+        if (isPaused()) {
+            revert Paused();
+        }
+
+        _;
+    }
+
+    /**
+     * @notice Reverts if the caller is not the pauser.
+     */
+    modifier onlyPauser() {
+        if (msg.sender != pauser()) {
+            revert OnlyPauser();
+        }
+
+        _;
+    }
+
+    /**
+     * @notice Pauses the contract.
+     */
+    function pause() public onlyPauser {
+        _setPaused(true);
+        emit ContractPaused();
+    }
+
+    /**
+     * @notice Unpauses the contract.
+     */
+    function unpause() public onlyPauser {
+        _setPaused(false);
+        emit ContractUnpaused();
+    }
+
+    /**
+     * @notice Returns whether or not the contract is paused.
+     * @return `true` if the contract is paused, `false` otherwise.
+     */
+    function isPaused() public view returns (bool) {
+        bool paused;
+        bytes32 pausedPosition = PAUSED_POSITION;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            paused := sload(pausedPosition)
+        }
+        return paused;
+    }
+
+    /**
+     * @notice Returns the address permissioned to pause/unpause this contract.
+     */
+    function pauser() public view returns (address) {
+        address pauserAddress;
+        bytes32 pauserPosition = PAUSER_POSITION;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            pauserAddress := sload(pauserPosition)
+        }
+        return pauserAddress;
+    }
+
+    /**
+     * @notice Sets the contract's paused state.
+     * @param paused `true` for paused, `false` for unpaused.
+     */
+    function _setPaused(bool paused) internal {
+        bytes32 pausedPosition = PAUSED_POSITION;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            sstore(pausedPosition, paused)
+        }
+    }
+
+    /**
+     * @notice Sets the address permissioned to pause this contract.
+     * @param _pauser The new pauser.
+     * @dev This shuold be wrapped by the inheriting contract, likely in a
+     * permissioned function like `onlyOwner`.
+     */
+    function _setPauser(address _pauser) internal {
+        bytes32 pauserPosition = PAUSER_POSITION;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            sstore(pauserPosition, _pauser)
+        }
+        emit PauserSet(_pauser);
+    }
```

Comment:
Don't see that as particularly useful. If it's an honest mistake, it can be easily fixed. If a malicious party were able to set this, it doesn't matter if they use an EOA or contract.

And leaving the option for EOA's is convenient, it's used in testing.

Reply:
Sounds good. 

What do you think about including a requirement that `_pauser` is a non zero address & different than the current one?

Reply:
+1 for adding checks for default values such as address(0)